QuickStart/ViewController.swift

```diff
@@ -0,0 +1,343 @@
+                                it is of the form https://<instance/<tenant>, where <instance> is the
+                                directory host (e.g. https://login.microsoftonline.com) and <tenant> is a
+                                identifier within the directory itself (e.g. a domain associated to the
+                                tenant, such as contoso.onmicrosoft.com, or the GUID representing the
+                                TenantID property of the directory)
+         - error                The error that occurred creating the application object, if any, if you're
+                                not interested in the specific error pass in nil.
+         */
+        
+        self.applicationContext = ADAuthenticationContext(authority: kAuthority, error: nil)
+        super.viewDidLoad()
+
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+
+        super.viewWillAppear(animated)
+        signoutButton.isEnabled = (currentAccount()?.accessToken != nil)
+    }
+    
+    /**
+     This button will invoke the authorization flow.
+    */
+
+    @IBAction func callGraphButton(_ sender: UIButton) {
+            
+        self.callAPI()
+
+    }
+
+    func acquireToken(completion: @escaping (_ success: Bool) -> Void) {
+        
+        guard let applicationContext = self.applicationContext else { return }
+
+        /**
+         
+         Acquire a token for an account
+         
+         - withResource:        The resource you wish to access. This will the Microsoft Graph API for this sample.
+         - clientId:            The clientID of your application, you should get this from the app portal.
+         - redirectUri:         The redirect URI that your application will listen for to get a response of the
+                                Auth code after authentication. Since this a native application where authentication
+                                happens inside the app itself, we can listen on a custom URI that the SDK knows to
+                                look for from within the application process doing authentication.
+         - completionBlock:     The completion block that will be called when the authentication
+         flow completes, or encounters an error.
+         */
+
+        applicationContext.acquireToken(withResource: kGraphURI, clientId: kClientID, redirectUri:kRedirectUri){ (result) in
+            
+        guard let result = result else {
+                
+                self.updateLogging(text: "Could not acquire token: No result returned")
+                return
+            }
+
+            if (result.status != AD_SUCCEEDED) {
+
+                if result.error.domain == ADAuthenticationErrorDomain
+                    && result.error.code == ADErrorCode.ERROR_UNEXPECTED.rawValue {
+                    
+                    self.updateLogging(text: "Unexpected internal error occured");
+                    completion(false)
```

Comment:
completion can move outside of this block here as the next block would do the same completion(false)